packages/flutter/test/cupertino/sliding_segmented_control_test.dart

```diff
@@ -0,0 +1,1091 @@
+    final Map<int, Widget> children = <int, Widget>{};
+    children[0] = Container(
+      constraints: const BoxConstraints.tightFor(height: 100.0),
+    );
+    children[1] = Container(
+      constraints: const BoxConstraints.tightFor(height: 400.0),
+    );
+    children[2] = Container(
+      constraints: const BoxConstraints.tightFor(height: 200.0),
+    );
+
+    final ValueNotifier<int> controller = ValueNotifier<int>(null);
+
+    await tester.pumpWidget(
+      boilerplate(
+        child: CupertinoSlidingSegmentedControl<int>(
+          key: const ValueKey<String>('Segmented Control'),
+          children: children,
+          controller: controller,
+        ),
+      ),
+    );
+
+    final RenderBox buttonBox = tester.renderObject(
+      find.byKey(const ValueKey<String>('Segmented Control')),
+    );
+
+    expect(
+      buttonBox.size.height,
+      400.0 + 2 * 2, // 2 px padding on both sides.
+    );
+  });
+
+  testWidgets('Width of each segmented control segment is determined by widest widget', (WidgetTester tester) async {
+    final Map<int, Widget> children = <int, Widget>{};
+    children[0] = Container(
+      constraints: const BoxConstraints.tightFor(width: 50.0),
+    );
+    children[1] = Container(
+      constraints: const BoxConstraints.tightFor(width: 100.0),
+    );
+    children[2] = Container(
+      constraints: const BoxConstraints.tightFor(width: 200.0),
+    );
+
+    final ValueNotifier<int> controller = ValueNotifier<int>(null);
+    await tester.pumpWidget(
+      boilerplate(
+        child: CupertinoSlidingSegmentedControl<int>(
+          key: const ValueKey<String>('Segmented Control'),
+          children: children,
+          controller: controller,
+        ),
+      ),
+    );
+
+    final RenderBox segmentedControl = tester.renderObject(
+      find.byKey(const ValueKey<String>('Segmented Control')),
+    );
+
+    // Subtract the 8.0px for horizontal padding separator. Remaining width should be allocated
+    // to each child equally.
+    final double childWidth = (segmentedControl.size.width - 8) / 3;
+
+    expect(childWidth, 200.0 + 9.25 * 2);
+  });
+
+  testWidgets('Width is finite in unbounded space', (WidgetTester tester) async {
+    final Map<int, Widget> children = <int, Widget>{};
+    children[0] = const Text('Child 1');
+    children[1] = const Text('Child 2');
+
+    final ValueNotifier<int> controller = ValueNotifier<int>(null);
+
+    await tester.pumpWidget(
+      boilerplate(
+        child: Row(
+          children: <Widget>[
+            CupertinoSlidingSegmentedControl<int>(
+              key: const ValueKey<String>('Segmented Control'),
+              children: children,
+              controller: controller,
+            ),
+          ],
+        ),
+      ),
+    );
+
+    final RenderBox segmentedControl = tester.renderObject(
+      find.byKey(const ValueKey<String>('Segmented Control')),
+    );
+
+    expect(segmentedControl.size.width.isFinite, isTrue);
```

Comment:
Couldn't we check the actual width of the segmented control here? Could simplify the expected width arithmetic by making the children SizedBox(width: 100) or something similar.